Laravel.gitignore

```diff
@@ -1,3 +1,15 @@
+#Exclude vendor libraries.
+#Including composer.lock and install using composer install.
+#Comment if you want to include all libraries
+/vendor
+
+#Node related
+/node_modules
+
+#Homestead related
+Homestead.yaml
```

Comment:
Can you tell me a bit about why these files can be safely ignored?

I can't see much details in the docs http://laravel.com/docs/5.1/homestead